web/profiles/joinup/joinup.behat.inc

```diff
@@ -105,28 +105,27 @@ class JoinupSubContext extends DrupalSubContextBase implements DrupalSubContextI
    */
   public function submitRegistrationForm(TableNode $table, $location) {
     $this->visitPath($location);
+    $username = '';
 
     // Fill in fields.
     foreach ($table->getRowsHash() as $field => $value) {
       $this->getSession()->getPage()->fillField($field, $value);
+      if ($field == 'Username') {
+        $username = $value;
+      }
     };
     // Submit form.
     $this->getSession()->getPage()->pressButton("Create new account");
 
     // Get the last registered user.
-    $storage = \Drupal::entityManager()->getStorage('user');
-    $query = $storage->getQuery();
-    $result = $query->sort('uid', 'DESC')->range(0, 1)->execute();
-    $uid = array_pop($result);
+    $results = \Drupal::entityManager()->getStorage('user')
+        ->loadByProperties(['name' => $username]);
```

Comment:
This code runs whenever someone uses this step:

```
@Given the following user registration at :location:
```

If you don't supply the username in the table then this method will not work as expected. That's fully the responsibility of this method.

I also don't really see the need for this step at all. Behat has everything built in to register a user through the UI. You just navigate to the page, fill in the fields, press the button and check that the success message is displayed. This is not good from a UAT perspective.